core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -408,8 +418,17 @@ public IndexShard relocated(String reason) throws IndexShardNotStartedException
             if (state != IndexShardState.STARTED) {
```

Comment:
we can remove the mutex here, but tbh this whole check can go away - It's just an extra safety mechanism. We can keep it simple